### PR TITLE
BUG ModelAdmin pagination with a filter

### PR DESCRIPTION
**Pull request by lukereative**
Fix silverstripe/silverstripe-framework#8454

### CODE REVIEW

File: code/ModelAdmin.php [hunk 1]
```diff
@@ -202,6 +203,9 @@ function ($form) {
                 $this->extend('updateSearchForm', $form);
             }
         ));
+        // GridFieldPaginator has to be added after filter header for it to function correctly
+        $listField->getConfig()->removeComponentsByType(GridFieldPaginator::class);
+        $listField->getConfig()->addComponent(new GridFieldPaginator());
```
Comment (robbieaverill) [inline code/ModelAdmin.php]: I guess we need some APIs to let people control GridField component priority, eh?

Comment (robbieaverill) [inline code/ModelAdmin.php]: Can you please use Injector to resolve this component?

Comment (lukereative) [reply code/ModelAdmin.php]: Done

Comment (lukereative) [reply code/ModelAdmin.php]: Yeah…

File: code/ModelAdmin.php [hunk 2]
```diff
@@ -202,6 +204,9 @@ function ($form) {
                 $this->extend('updateSearchForm', $form);
             }
         ));
+        // GridFieldPaginator has to be added after filter header for it to function correctly
+        $listField->getConfig()->removeComponentsByType(GridFieldPaginator::class);
+        $listField->getConfig()->addComponent(Injector::inst()->get(GridFieldPaginator::class));
```
Comment (tractorcow) [inline code/ModelAdmin.php]: This ignores the `page_length` config for modeladmin. You should be using `create` for items that have member properties, and `get` for global services / singletons.

**Pull request merged**